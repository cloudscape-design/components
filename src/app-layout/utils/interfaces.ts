// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import React from 'react';

export interface SizeControlProps {
  position: 'side' | 'bottom';
  panelRef?: React.RefObject<HTMLDivElement>;
  handleRef?: React.RefObject<HTMLDivElement>;
  onResize: (newSize: number) => void;
  hasTransitions?: boolean;
}
