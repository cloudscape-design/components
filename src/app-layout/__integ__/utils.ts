// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

export const testIf = (condition: boolean) => (condition ? test : test.skip);

export type Theme = 'classic' | 'refresh' | 'refresh-toolbar';

export function getUrlParams(theme: Theme) {
  const params = new URLSearchParams({
    visualRefresh: `${theme.startsWith('refresh')}`,
    appLayoutToolbar: `${theme === 'refresh-toolbar'}`,
  });
  return params.toString();
}
