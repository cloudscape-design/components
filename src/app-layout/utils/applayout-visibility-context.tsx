// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import { createContext } from 'react';

export const AppLayoutVisibilityContext = createContext<boolean>(true);
