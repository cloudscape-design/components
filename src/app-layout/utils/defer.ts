// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
export class Deferred<T> {
  private readonly _promise: Promise<T>;
  private _resolve?: (value?: any) => void;
  private _reject?: (reason?: any) => void;

  constructor() {
    this._promise = new Promise<T>((resolve, reject) => {
      this._resolve = resolve;
      this._reject = reject;
    });
  }

  get promise(): Promise<T> {
    return this._promise;
  }

  resolve = (value?: T | PromiseLike<T>): void => {
    this._resolve?.(value);
  };

  reject = (reason?: any): void => {
    this._reject?.(reason);
  };
}
