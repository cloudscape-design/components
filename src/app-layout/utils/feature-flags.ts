// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import { getGlobalFlag } from '@cloudscape-design/component-toolkit/internal';

import { useVisualRefresh } from '../../internal/hooks/use-visual-mode';

export const useAppLayoutToolbarEnabled = () => {
  const isRefresh = useVisualRefresh();
  return isRefresh && (getGlobalFlag('appLayoutWidget') || getGlobalFlag('appLayoutToolbar'));
};
