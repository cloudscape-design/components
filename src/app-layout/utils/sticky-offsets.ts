// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import globalVars from '../../internal/styles/global-vars';

export function getStickyOffsetVars(
  headerHeight: number,
  footerHeight: number,
  stickyNotificationsHeight: string,
  mobileToolbarHeight: string,
  disableBodyScroll: boolean,
  isMobile: boolean,
  extraTopOffset?: string
) {
  return {
    [globalVars.stickyVerticalTopOffset]: `calc(${!disableBodyScroll ? headerHeight : 0}px + ${
      isMobile ? mobileToolbarHeight : stickyNotificationsHeight
    } + ${extraTopOffset ?? '0px'})`,
    [globalVars.stickyVerticalBottomOffset]: `${!disableBodyScroll ? footerHeight : 0}px`,
  };
}
