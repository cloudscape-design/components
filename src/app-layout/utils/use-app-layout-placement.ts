// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import { useCallback, useEffect, useRef, useState } from 'react';
import { useObservedElement } from './use-observed-element';
import { useResizeObserver } from '@cloudscape-design/component-toolkit/internal';
import { getLogicalBoundingClientRect } from '../../internal/direction';

export function useAppLayoutPlacement(headerSelector: string, footerSelector: string) {
  const mainElementRef = useRef<HTMLElement>(null);
  const headerHeight = useObservedElement(headerSelector);
  const footerHeight = useObservedElement(footerSelector);
  const [offsets, setOffsets] = useState({
    insetInlineStart: 0,
    insetInlineEnd: 0,
    inlineSize: Number.POSITIVE_INFINITY,
  });

  const updatePosition = useCallback(() => {
    if (!mainElementRef.current) {
      return;
    }
    const { insetInlineStart, insetInlineEnd, inlineSize } = getLogicalBoundingClientRect(mainElementRef.current);

    // skip reading sizes in JSDOM
    if (inlineSize === 0) {
      return;
    }

    setOffsets({ insetInlineStart, insetInlineEnd: inlineSize - insetInlineEnd, inlineSize });
  }, []);

  useEffect(() => {
    window.addEventListener('resize', updatePosition);
    return () => window.removeEventListener('resize', updatePosition);
  }, [updatePosition]);

  useResizeObserver(mainElementRef, updatePosition);

  return [mainElementRef, { ...offsets, insetBlockStart: headerHeight, insetBlockEnd: footerHeight }] as const;
}
