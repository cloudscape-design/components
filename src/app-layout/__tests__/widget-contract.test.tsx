// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import React from 'react';
import { render } from '@testing-library/react';

import AppLayout from '../../../lib/components/app-layout';
import { describeEachAppLayout, testDrawer } from './utils';

const renderedProps = new Map();
function createWidgetizedComponentMock(Implementation: React.ComponentType) {
  return () => {
    return function Widgetized(props: any) {
      renderedProps.set(Implementation, props);
      return null;
    };
  };
}

jest.mock('../../../lib/components/internal/widgets', () => ({
  createWidgetizedComponent: createWidgetizedComponentMock,
}));
jest.mock('@cloudscape-design/component-toolkit/internal', () => {
  let counter = 0;
  return {
    ...jest.requireActual('@cloudscape-design/component-toolkit/internal'),
    useUniqueId: (prefix: string) => `${prefix}${++counter}`,
  };
});

describeEachAppLayout({ themes: ['refresh-toolbar'], sizes: ['desktop'] }, () => {
  beforeEach(() => {
    renderedProps.clear();
  });

  test('contract for default use-case', () => {
    render(<AppLayout />);
    expect(renderedProps).toMatchSnapshot();
  });

  test('contract with all slots provided', () => {
    render(
      <AppLayout
        breadcrumbs={<div>breadcrumbs</div>}
        notifications={<div>notifications</div>}
        navigation={<div>navigation</div>}
        tools={<div>tools</div>}
        content={<div>content</div>}
        splitPanel={<div>split panel</div>}
      />
    );
    expect(renderedProps).toMatchSnapshot();
  });

  test('contract with drawers', () => {
    render(<AppLayout activeDrawerId={testDrawer.id} onDrawerChange={() => {}} drawers={[testDrawer]} />);
    expect(renderedProps).toMatchSnapshot();
  });
});
