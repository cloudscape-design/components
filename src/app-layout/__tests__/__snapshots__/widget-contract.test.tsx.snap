// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme=refresh-toolbar, Size=desktop contract for default use-case 1`] = `
Map {
  "BeforeMainSlotImplementation" => {
    "appLayoutProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "contentType": "default",
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigationOpen": true,
      "navigationWidth": 280,
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "toolsWidth": 290,
    },
    "appLayoutState": {
      "isIntersecting": true,
    },
    "toolbarProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "hasNavigation": true,
      "navigationFocusRef": undefined,
      "navigationOpen": true,
      "onNavigationToggle": undefined,
    },
  },
  "TopContentSlotImplementation" => {
    "appLayoutProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "contentType": "default",
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigationOpen": true,
      "navigationWidth": 280,
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "toolsWidth": 290,
    },
    "appLayoutState": {
      "isIntersecting": true,
    },
    "toolbarProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "hasNavigation": true,
      "navigationFocusRef": undefined,
      "navigationOpen": true,
      "onNavigationToggle": undefined,
    },
  },
  "BottomContentSlotImplementation" => {
    "appLayoutProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "contentType": "default",
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigationOpen": true,
      "navigationWidth": 280,
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "toolsWidth": 290,
    },
    "appLayoutState": {
      "isIntersecting": true,
    },
    "toolbarProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "hasNavigation": true,
      "navigationFocusRef": undefined,
      "navigationOpen": true,
      "onNavigationToggle": undefined,
    },
  },
  "AfterMainSlotImplementation" => {
    "appLayoutProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "contentType": "default",
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigationOpen": true,
      "navigationWidth": 280,
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "toolsWidth": 290,
    },
    "appLayoutState": {
      "isIntersecting": true,
    },
    "toolbarProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "hasNavigation": true,
      "navigationFocusRef": undefined,
      "navigationOpen": true,
      "onNavigationToggle": undefined,
    },
  },
  "AppLayoutStateProvider" => {
    "appLayoutProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "contentType": "default",
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigationOpen": true,
      "navigationWidth": 280,
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "toolsWidth": 290,
    },
    "forwardRef": null,
    "stateManager": {
      "current": {
        "hasToolbar": true,
        "setState": [Function],
        "setToolbar": undefined,
      },
    },
  },
}
`;

exports[`Theme=refresh-toolbar, Size=desktop contract with all slots provided 1`] = `
Map {
  "BeforeMainSlotImplementation" => {
    "appLayoutProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "breadcrumbs": <div>
        breadcrumbs
      </div>,
      "content": <div>
        content
      </div>,
      "contentType": "default",
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigation": <div>
        navigation
      </div>,
      "navigationOpen": true,
      "navigationWidth": 280,
      "notifications": <div>
        notifications
      </div>,
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "splitPanel": <div>
        split panel
      </div>,
      "tools": <div>
        tools
      </div>,
      "toolsWidth": 290,
    },
    "appLayoutState": {
      "isIntersecting": true,
    },
    "toolbarProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "hasBreadcrumbsPortal": true,
      "hasNavigation": true,
      "navigationFocusRef": undefined,
      "navigationOpen": true,
      "onNavigationToggle": undefined,
    },
  },
  "TopContentSlotImplementation" => {
    "appLayoutProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "breadcrumbs": <div>
        breadcrumbs
      </div>,
      "content": <div>
        content
      </div>,
      "contentType": "default",
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigation": <div>
        navigation
      </div>,
      "navigationOpen": true,
      "navigationWidth": 280,
      "notifications": <div>
        notifications
      </div>,
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "splitPanel": <div>
        split panel
      </div>,
      "tools": <div>
        tools
      </div>,
      "toolsWidth": 290,
    },
    "appLayoutState": {
      "isIntersecting": true,
    },
    "toolbarProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "hasBreadcrumbsPortal": true,
      "hasNavigation": true,
      "navigationFocusRef": undefined,
      "navigationOpen": true,
      "onNavigationToggle": undefined,
    },
  },
  "BottomContentSlotImplementation" => {
    "appLayoutProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "breadcrumbs": <div>
        breadcrumbs
      </div>,
      "content": <div>
        content
      </div>,
      "contentType": "default",
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigation": <div>
        navigation
      </div>,
      "navigationOpen": true,
      "navigationWidth": 280,
      "notifications": <div>
        notifications
      </div>,
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "splitPanel": <div>
        split panel
      </div>,
      "tools": <div>
        tools
      </div>,
      "toolsWidth": 290,
    },
    "appLayoutState": {
      "isIntersecting": true,
    },
    "toolbarProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "hasBreadcrumbsPortal": true,
      "hasNavigation": true,
      "navigationFocusRef": undefined,
      "navigationOpen": true,
      "onNavigationToggle": undefined,
    },
  },
  "AfterMainSlotImplementation" => {
    "appLayoutProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "breadcrumbs": <div>
        breadcrumbs
      </div>,
      "content": <div>
        content
      </div>,
      "contentType": "default",
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigation": <div>
        navigation
      </div>,
      "navigationOpen": true,
      "navigationWidth": 280,
      "notifications": <div>
        notifications
      </div>,
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "splitPanel": <div>
        split panel
      </div>,
      "tools": <div>
        tools
      </div>,
      "toolsWidth": 290,
    },
    "appLayoutState": {
      "isIntersecting": true,
    },
    "toolbarProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "hasBreadcrumbsPortal": true,
      "hasNavigation": true,
      "navigationFocusRef": undefined,
      "navigationOpen": true,
      "onNavigationToggle": undefined,
    },
  },
  "AppLayoutStateProvider" => {
    "appLayoutProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "breadcrumbs": <div>
        breadcrumbs
      </div>,
      "content": <div>
        content
      </div>,
      "contentType": "default",
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigation": <div>
        navigation
      </div>,
      "navigationOpen": true,
      "navigationWidth": 280,
      "notifications": <div>
        notifications
      </div>,
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "splitPanel": <div>
        split panel
      </div>,
      "tools": <div>
        tools
      </div>,
      "toolsWidth": 290,
    },
    "forwardRef": null,
    "stateManager": {
      "current": {
        "hasToolbar": true,
        "setState": [Function],
        "setToolbar": undefined,
      },
    },
  },
}
`;

exports[`Theme=refresh-toolbar, Size=desktop contract with drawers 1`] = `
Map {
  "BeforeMainSlotImplementation" => {
    "appLayoutProps": {
      "activeDrawerId": "security",
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "contentType": "default",
      "drawers": [
        {
          "ariaLabels": {
            "closeButton": "Security close button",
            "drawerName": "Security drawer content",
            "resizeHandle": "Security resize handle",
            "triggerButton": "Security trigger button",
          },
          "content": <span>
            Security
          </span>,
          "id": "security",
          "trigger": {
            "iconName": "security",
          },
        },
      ],
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigationOpen": true,
      "navigationWidth": 280,
      "onDrawerChange": [Function],
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "toolsWidth": 290,
    },
    "appLayoutState": {
      "isIntersecting": true,
    },
    "toolbarProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "hasNavigation": true,
      "navigationFocusRef": undefined,
      "navigationOpen": true,
      "onNavigationToggle": undefined,
    },
  },
  "TopContentSlotImplementation" => {
    "appLayoutProps": {
      "activeDrawerId": "security",
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "contentType": "default",
      "drawers": [
        {
          "ariaLabels": {
            "closeButton": "Security close button",
            "drawerName": "Security drawer content",
            "resizeHandle": "Security resize handle",
            "triggerButton": "Security trigger button",
          },
          "content": <span>
            Security
          </span>,
          "id": "security",
          "trigger": {
            "iconName": "security",
          },
        },
      ],
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigationOpen": true,
      "navigationWidth": 280,
      "onDrawerChange": [Function],
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "toolsWidth": 290,
    },
    "appLayoutState": {
      "isIntersecting": true,
    },
    "toolbarProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "hasNavigation": true,
      "navigationFocusRef": undefined,
      "navigationOpen": true,
      "onNavigationToggle": undefined,
    },
  },
  "BottomContentSlotImplementation" => {
    "appLayoutProps": {
      "activeDrawerId": "security",
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "contentType": "default",
      "drawers": [
        {
          "ariaLabels": {
            "closeButton": "Security close button",
            "drawerName": "Security drawer content",
            "resizeHandle": "Security resize handle",
            "triggerButton": "Security trigger button",
          },
          "content": <span>
            Security
          </span>,
          "id": "security",
          "trigger": {
            "iconName": "security",
          },
        },
      ],
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigationOpen": true,
      "navigationWidth": 280,
      "onDrawerChange": [Function],
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "toolsWidth": 290,
    },
    "appLayoutState": {
      "isIntersecting": true,
    },
    "toolbarProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "hasNavigation": true,
      "navigationFocusRef": undefined,
      "navigationOpen": true,
      "onNavigationToggle": undefined,
    },
  },
  "AfterMainSlotImplementation" => {
    "appLayoutProps": {
      "activeDrawerId": "security",
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "contentType": "default",
      "drawers": [
        {
          "ariaLabels": {
            "closeButton": "Security close button",
            "drawerName": "Security drawer content",
            "resizeHandle": "Security resize handle",
            "triggerButton": "Security trigger button",
          },
          "content": <span>
            Security
          </span>,
          "id": "security",
          "trigger": {
            "iconName": "security",
          },
        },
      ],
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigationOpen": true,
      "navigationWidth": 280,
      "onDrawerChange": [Function],
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "toolsWidth": 290,
    },
    "appLayoutState": {
      "isIntersecting": true,
    },
    "toolbarProps": {
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "hasNavigation": true,
      "navigationFocusRef": undefined,
      "navigationOpen": true,
      "onNavigationToggle": undefined,
    },
  },
  "AppLayoutStateProvider" => {
    "appLayoutProps": {
      "activeDrawerId": "security",
      "ariaLabels": {
        "drawers": undefined,
        "drawersOverflow": undefined,
        "drawersOverflowWithBadge": undefined,
        "navigation": undefined,
        "navigationClose": undefined,
        "navigationToggle": undefined,
        "notifications": undefined,
        "tools": undefined,
        "toolsClose": undefined,
        "toolsToggle": undefined,
      },
      "contentType": "default",
      "drawers": [
        {
          "ariaLabels": {
            "closeButton": "Security close button",
            "drawerName": "Security drawer content",
            "resizeHandle": "Security resize handle",
            "triggerButton": "Security trigger button",
          },
          "content": <span>
            Security
          </span>,
          "id": "security",
          "trigger": {
            "iconName": "security",
          },
        },
      ],
      "maxContentWidth": undefined,
      "minContentWidth": 280,
      "navigationOpen": true,
      "navigationWidth": 280,
      "onDrawerChange": [Function],
      "onNavigationChange": [Function],
      "placement": {
        "inlineSize": Infinity,
        "insetBlockEnd": 0,
        "insetBlockStart": 0,
        "insetInlineEnd": 0,
        "insetInlineStart": 0,
      },
      "toolsWidth": 290,
    },
    "forwardRef": null,
    "stateManager": {
      "current": {
        "hasToolbar": true,
        "setState": [Function],
        "setToolbar": undefined,
      },
    },
  },
}
`;
